sdk/src/main/java/io/radar/sdk/Radar.kt

```diff
@@ -852,6 +841,89 @@ object Radar {
         })
     }
 
+    /**
+     * Tracks the user's location with device integrity information for location verification use cases.
+     *
+     * Note that you must call `setVerifiedHost()` and configure SSL pinning before calling this method.
+     *
+     * @see [](https://radar.com/documentation/fraud)
+     */
+    @JvmStatic
+    fun trackVerified(callback: RadarTrackCallback? = null) {
+        if (!initialized) {
+            callback?.onComplete(RadarStatus.ERROR_PUBLISHABLE_KEY)
+
+            return
+        }
+
+        if (!this::verificationManager.isInitialized) {
+            this.verificationManager = RadarVerificationManager(this.context, this.logger)
+        }
+
+        apiClient.getConfig(object : RadarApiClient.RadarGetConfigApiCallback {
+            override fun onComplete(config: RadarConfig) {
+                locationManager.getLocation(RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.HIGH, RadarLocationSource.FOREGROUND_LOCATION, object : RadarLocationCallback {
+                    override fun onComplete(status: RadarStatus, location: Location?, stopped: Boolean) {
+                        if (status != RadarStatus.SUCCESS || location == null) {
+                            handler.post {
+                                callback?.onComplete(status)
+                            }
+
+                            return
+                        }
+
+                        verificationManager.getIntegrityToken(config.googlePlayProjectNumber, config.nonce) { integrityToken, integrityException ->
+                            apiClient.track(location, RadarState.getStopped(context), RadarActivityLifecycleCallbacks.foreground, RadarLocationSource.FOREGROUND_LOCATION, false, null, true, integrityToken, integrityException, object : RadarApiClient.RadarTrackApiCallback {
+                                override fun onComplete(
+                                    status: RadarStatus,
+                                    res: JSONObject?,
+                                    events: Array<RadarEvent>?,
+                                    user: RadarUser?,
+                                    nearbyGeofences: Array<RadarGeofence>?,
+                                    config: RadarConfig?,
+                                ) {
```

Comment:
@jkao suggested we add a single retry if the call to `.track` fails due to a `verificationRequired` error, in INF-245. WDYT? Would be interested to see what the best way to implement a retry would be.

Reply:
Jeff said that the caller could handle this. Marking resolved.